src/plan_expand_hypertable.c

```diff
@@ -51,54 +92,67 @@ collect_quals_walker(Node *node, CollectQualCtx *ctx)
 		{
 			Node	   *qual = (Node *) lfirst(lc);
 			RestrictInfo *restrictinfo;
+			Relids		relids;
+			bool		belongs_to_cur_rel = false;
 
-			Relids		relids = pull_varnos(qual);
+			relids = pull_varnos(qual);
 
-			if (bms_num_members(relids) != 1 || !bms_is_member(ctx->rel->relid, relids))
+			belongs_to_cur_rel = bms_num_members(relids) == 1 && bms_is_member(ctx->rel->relid, relids);
+
+			if (!belongs_to_cur_rel)
 				continue;
+
+			if (is_chunk_exclusion_func(qual))
+			{
+				FuncExpr   *func_expr = (FuncExpr *) qual;
+
+				/* validation */
+				if (ctx->chunk_exclusion_func != NULL && belongs_to_cur_rel)
+					ereport(ERROR,
+							(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+							 errmsg("only one chunks_in call is allowed per hypertable")));
+
+				Assert(func_expr->args->length == 2);
+				if (!IsA(linitial(func_expr->args), Var))
+					ereport(ERROR,
+							(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+							 errmsg("first parameter for chunks_in function needs to be a relation or alias")));
```

Comment:
Also, is checking that it is a `Var` enough here? Don't we need to check, e.g., `varno` and `vartype` to make sure the `Var` corresponds to the correct relation?

Reply:
Again I am not sure what's the best name here. To me it seems that relation/alias is more user friendly. How you recommend to name this? `Row` seems a bit artificial to me?

Regarding `Var` checking, prior to this we actually made sure that `Var` belongs to current relation. No need to check var type since function signature restricts it to `Record`.

Reply:
You are requiring a row reference here, because you could still pass in a row that isn't a reference.
SELECT chunks_in(ROW(1,2),ARRAY[1,2,3]);
should probably add that as test